system/jlib/jsocket.cpp

```diff
@@ -4230,6 +4231,9 @@ class CSocketSelectThread: public CSocketBaseThread
                     n++;
             }
         }
+        // seems this should be 1/4 or 1/8 or so
+        // of the overall XFD_SETSIZE fd maximum
+        // to get multiple select handler threads
```

Comment:
Why? Afaik, the idea was that it used FD_SETSIZE fully before spinning off another thread that could also use another FD_SETSIZE worth off handles.

Reply:
I understand now.
The jsocket code in general (not just this handler) in the _USE_SELECT branches can't cope with handle *numbers* > than FD_SETSIZE, because of the way the set bits are manipulated that depend on FD_SETSIZE.

It _could_ have span off new thread on some other lower limit as long as highest file handler # was less than FD_SETSIZE, and XFD_SETSIZE could probably have been increased significantly as long as ::select wasn't dealing with an excessively large set which makes it very slow.

Anyway, this is all academic now, since the select code is legacy and should be removed completely at some point.

Reply:
Yes, bitmask width limits fd value with select() - which is why epoll() and poll() are now used everywhere by default.